### PR TITLE
feature delete user account

### PR DESCRIPTION
**Pull request by Amean**
## What kind of change does this PR introduce?

Adds delete user account feature

## What is the current behavior?

unable to delete user account

Please link any relevant issues     #25 

## What is the new behavior?

WIP having some permissions issues

### CODE REVIEW

**Comment by MateusAmin** [comment]
How does the authentication work with the Service Role Key?

**Comment by MateusAmin** [comment]
https://app.supabase.io/project/amsbkonxeeapffdsedwm/api/default?page=auth

**Comment by MateusAmin** [comment]
Got it working.

**Comment by MateusAmin** [comment]
example:

```dart
final supabaseClient = SupabaseClient(supaBaseUrl, serviceRowKey);

deleteUserReply = await supabaseClient.auth.api.deleteUser(
        serviceRowKey,
        userid,
      );
```

**Comment by dshukertjr** [comment]
Hi @MateusAmin !

Thank you so much for this PR! Your work is amazing!

Just curious to hear what the use case for this is. Are you currently using Dart as your backend?

**Comment by MateusAmin** [comment]
@dshukertjr Thanks for looking. Replied in the issue: https://github.com/supabase/gotrue-dart/pulls#issuecomment-881981600

File: lib/src/fetch.dart [hunk 1]
```diff
@@ -103,4 +103,29 @@ class Fetch {
       client.close();
     }
   }
+
```
Comment (bdlukaa) [inline lib/src/fetch.dart:106]:
```suggestion

  /// Delete an user.
  ///
  /// `uid` is the user uid you want to remove.
  ///
  /// `jwt`: A valid JWT. Must be a full-access API key (e.g. service_role key).
```

**Comment by bdlukaa** [comment]
LGMT! Since the method is not exposed in `GoTrueClient` and a `jwt` key is necessary.

**Comment by Tomic-Riedel** [comment]
Hey!

Thank you for your amazing work Amean!

Even if this pull request is still not approved, I wonder how to implement this into my Flutter app. I don't mean, how to call this function, I mean how to edit the package on my local storage and connect this local package then with  the supabase-flutter library?

Thanks in advanced!

**Comment by Vinzent03** [comment]
I think this can be reopened since #83 is merged.

**Comment by dshukertjr** [comment]
@Vinzent03 Yup, certainly! I think we should add all the server side apis to the dart library!

**Comment by Vinzent03** [comment]
I implemented that now in #106, so I'll close this

**Pull request closed without merging**